File: src/main/java/ExprParser/ExprTreeRange.java
package ExprParser;

import jAADD.AADD;
import static jAADD.AADDMgr.REAL;

/**
 * Predefined functions: range, uses AADD with new noise symbol.
 */
class ExprTreeRange extends ExprTreeFunction {
    ExprTreeRange() {
        super(REAL, "range",
                new ExprTree("min", REAL),
                new ExprTree("max", REAL));
    }
    void evalThis() throws ExprError {
        if (param.size() != 2) throw new ExprError("range expects 2 parameters");
        double l = getAADDParam(0).Value().getMin();
        double r = getAADDParam(1).Value().getMax();
        result = new AADD(l, r, -1);
    }
}

File: src/main/kotlin/com/github/tuk-cps/exprParser/ExprTreeRange.kt
package exprParser

import jAADD.AADD

/**
 * Predefined function: range, uses AADD with new noise symbol.
 */
internal class ExprTreeRange : ExprTreeFunction(AADD.Reals, "range", ExprTree(AADD.Reals, "min"), ExprTree(AADD.Reals, "max")) {
    @Throws(ExprError::class)
    override fun evalUp() {
        if (param.size != 2) throw ExprError("range expects 2 parameters")
        val l = getAADDParam(0).value!!.min
        val r = getAADDParam(1).value!!.max
        value = AADD.range(l, r, -1)
    }
}